### PR TITLE
refactor: lua code

### PR DESCRIPTION
**Pull request by altermo**
Mostly refactored Lua code, though a few bug fixes are sprinkled in.

By the way, which version of neovim is this plugin intended for?

### CODE REVIEW

File: lua/codeium/api.lua [hunk 1]
```diff
@@ -168,8 +159,7 @@ function Server:new()
 	function m.start()
 		m.shutdown()
 
-		local cookie = next_cookie()
```
Comment (pqn) [inline lua/codeium/api.lua:171]: Would you be able to explain the cookie change here?

File: lua/codeium/views/auth-menu.lua [hunk 2]
```diff
@@ -69,7 +61,6 @@ local function open_buffer(url, callback)
 	}
 
 	win_id = vim.api.nvim_open_win(bufnr, true, win_opts)
-	vim.api.nvim_win_set_option(win_id, "cursorline", false)
```
Comment (pqn) [inline lua/codeium/views/auth-menu.lua:72]: What's this deletion for?

**Review by pqn — APPROVED** [review]
Apologies for the delayed review. Would you be able to resolve the merge conflicts?

**Comment by pqn** [comment]
> By the way, which version of neovim is this plugin intended for?

I'm not sure as we are not the original authors of this plugin.

File: lua/codeium/views/auth-menu.lua (hunk 2 above)
Comment (altermo) [reply lua/codeium/views/auth-menu.lua:72]: `win_opts` contains an option `style=minimal` which disables `cursorline` in the window, so no need to do it again

**Comment by pqn** [comment]
@rain9441 do you think you could open a PR for your changes?

**Pull request merged**